src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1086,11 +1086,10 @@ protected override bool IsValidPath(string path)
 
             // Remove drive root first
             string pathWithoutDriveRoot = path.Substring(Path.GetPathRoot(path).Length);
-            char[] invalidFileChars = Path.GetInvalidFileNameChars();
 
             foreach (string segment in pathWithoutDriveRoot.Split(Path.DirectorySeparatorChar))
             {
-                if (segment.IndexOfAny(invalidFileChars) != -1)
+                if (PathUtils.ContainsInvalidFileNameChars(segment))
```

Comment:
I see we never check full path so we check more short names. Could you please update perf test with loop like this (split path)?

Reply:
@iSazonov Yep good point. I have updated bench mark in description. I've grouped by category filename/path and params (10, 100, 1000) paths so we can bench mark small & large number of paths. 

It seems pretty consistent with SearchValues outperforming IndexOfAny with char array.

Reply:
@iSazonov Very nice, I think I should save this benchmark somewhere 😄.

I've updated results. Let me know if anything else is needed, I think this benchmark is quite comprehensive.